src/libcore/slice/mod.rs

```diff
@@ -359,6 +359,10 @@ impl<T> [T] {
     /// The caller must ensure that the slice outlives the pointer this
     /// function returns, or else it will end up pointing to garbage.
     ///
+    /// The caller must also ensure that the memory the pointer (non-transitively) points to
+    /// is never written to (except inside an `UnsafeCell`). If you need to mutate
```

Comment:
Unsure if this qualifies as too pedantic or not, but "is never written to" sounds like it'd include the time after you stop using this pointer and any (raw?) reborrows thereof, the same way that (iiuc) constructing a Pin must guarantee the underlying value is never moved again for the rest of the program including after the Pin itself goes away.

Would "is never written to _using this pointer_" be better, or wrong in other ways?

Reply:
> Would "is never written to using this pointer" be better, or wrong in other ways?

"using this pointer or any pointer derived from it" would be correct and should cover your case?